From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martijn Muijsers <martijnmuijsers@live.nl>
Date: Mon, 8 Jan 2024 19:24:33 +0100
Subject: [PATCH] Modifiable Bukkit enums - Complete legacy conversion -
 Conversion is explicitly for item or block - Expose Bukkit API

Because converting legacy values is ambiguous when not indicating whether it concerns an item or block,
the "Modifiable Bukkit enums - Complete legacy conversion - Conversion is explicitly for item or block" patch in the `fiddle-server` module
requires that all such conversions explicitly indicate this.

To communicate this to plugin authors, this patch deprecates the `UnsafeValues.fromLegacy` methods that do not have an `itemPriority` parameter,
and adds a new method `fromLegacy(Material, boolean)` because it did not exist.

License: GPL-3.0 (https://www.gnu.org/licenses/gpl-3.0.html)
Fiddle - https://fiddlemc.org

diff --git a/src/main/java/org/bukkit/UnsafeValues.java b/src/main/java/org/bukkit/UnsafeValues.java
index e4c0862a848aaa99c174afde43ad46202b46435c..c4f9d4ab205f3a864ff525359dde17e5d9fd12b8 100644
--- a/src/main/java/org/bukkit/UnsafeValues.java
+++ b/src/main/java/org/bukkit/UnsafeValues.java
@@ -38,8 +38,23 @@ public interface UnsafeValues {
     void reportTimings(); // Paper
     Material toLegacy(Material material);
 
+
+    // Fiddle start - modifiable Bukkit enums - Material - make legacy conversion complete
+    /**
+     * @deprecated Use {@link #fromLegacy(Material, boolean)} instead.
+     */
+    @Deprecated
+    // Fiddle end - modifiable Bukkit enums - Material - make legacy conversion complete
     Material fromLegacy(Material material);
 
+    // Fiddle start - modifiable Bukkit enums - Material - make legacy conversion complete
+    Material fromLegacy(Material material, boolean itemPriority);
+
+    /**
+     * @deprecated Use {@link #fromLegacy(MaterialData, boolean)} instead.
+     */
+    @Deprecated
+    // Fiddle end - modifiable Bukkit enums - Material - make legacy conversion complete
     Material fromLegacy(MaterialData material);
 
     Material fromLegacy(MaterialData material, boolean itemPriority);
